dateutil/tokenization.py

```diff
@@ -0,0 +1,399 @@
+
+        If the optional argument validate (which defaults to True) is set to
+        False, then the function will not check that the hour/minute/second
+        values are valid, and will allow through e.g. "22:64".
+
+        return timelike
+        
+
+        >>> tup = ('02', ':', '61.45',)
+        >>> Window.is_timelike(tup, validate=False)
+        True
+
+        >>> tup = ('d2', ':', '47',)
+        >>> Window.is_timelike(tup, validate=True)
+        False
+
+        >>> tup = ('2', '-', '47',)
+        >>> Window.is_timelike(tup, validate=True)
+        False
+
+        >>> tup = ('0.2', ':', '47.0',)
+        >>> Window.is_timelike(tup, validate=True)
+        False
+
+        >>> tup = ('25', ':', '47',)
+        >>> Window.is_timelike(tup, hour_only=True, validate=True)
+        False
+
+        >>> tup = ('25', ':', '47',)
+        >>> Window.is_timelike(tup, hour_only=False, validate=True)
+        True
+
+        >>> tup = ('25', ':', '4',)
+        >>> Window.is_timelike(tup, hour_only=False, validate=True)
+        False
+
+        >>> tup = ('25', ':', '488',)
+        >>> Window.is_timelike(tup, hour_only=False, validate=True)
+        False
+
+        >>> tup = ('25', ':', '48t8',)
+        >>> Window.is_timelike(tup, hour_only=False, validate=True)
+        False
+
+        >>> tup = ('25', ':', '48.d',)
+        >>> Window.is_timelike(tup, hour_only=False, validate=True)
+        False
+
+        >>> tup = ('23', ':', '48.0',)
+        >>> Window.is_timelike(tup, hour_only=False, validate=True)
+        True
+
+        """
+        if len(tup) != 3:
+            return False
+            
+        (HM, colon, MS) = tup
+        # HM --> Hour or Minute
+        # MS --> Minute or Second
+        # colon --> : between HH:MM or MM:SS
+        
+        joined = ''.join(tup)
+        if not validate and re.search('^\d{1,2}:\d{2}([,\.]\d+)?$', joined):
+            return True
+
+        if colon != ':':
+            return False
+        elif not (HM.isdigit() and int(HM) <= 59):
+            return False
+        elif hour_only and not (int(HM) <= 23 and len(MS) == 2):
+            return False
+
+        integral = MS[:2]
+        fractional = MS[3:]
+
+        if not (integral.isdigit() and
+                len(integral) == 2 and int(integral) <= 59):
+            return False
+
+        elif len(MS) == 2:
+            # No fractional part, so this checks out.
+            return True
+            
+        elif MS[2] not in ['.', ',']:
+            # We allow for commas for internationalization and for
+            # python's logging format.  Anything else indicates a
+            # non-minute-or-second
+            return False
+
+        elif not fractional.isdigit():
+            # Note that ''.isdigit() returns False, so this handles
+            # the case of a trailing "." or ","
+            return False
+
+        return True
+
+
+_inf = float('INF')
+_minf = float('-INF')
+def check_float_like(item):
```

Comment:
This is some non-date-related utility function. Definitely not a good fit for dateutil.

Reply:
This func is used for the added test case `test_misleading_float`.